lib/videojs_rails/tags/source.rb

```diff
@@ -2,37 +2,28 @@ module VideojsRails
   module Tags
     class Source < Tag
       def initialize(video_type, options)
-        @video_type = video_type
-        parse_options(options)
+        @attributes = prepare_attributes(video_type, options)
       end
 
       def to_html
-        tag :source, attributes
+        tag :source, @attributes
       end
 
       private
 
-      attr_reader :video_type, :src
-
-      def parse_options(options)
+      def prepare_attributes(video_type, options)
         case options
-        when String, Symbol
-          @src = options
-        else
-          raise ArgumentError
+          when String, Symbol
+            {}.tap do |attributes|
```

Comment:
Why `tap`? In that case you can just return:

~~~
{
	src: options
	type: "video/#{video_type}"
}
~~~